product-mini/platforms/zephyr/simple/build_and_run.sh

```diff
@@ -17,11 +16,10 @@ QEMU_ARC_TARGET="qemu_arc"
 usage ()
 {
         echo "USAGE:"
-        echo "$0 $X86_TARGET|$STM32_TARGET|$ESP32_TARGET|$ESP32C3_TARGET|$PARTICLE_ARGON_TARGET|$QEMU_CORTEX_A53|$QEMU_XTENSA_TARGET|$QEMU_RISCV64_TARGET|$QEMU_RISCV32_TARGET|$QEMU_ARC_TARGET"
+        echo "$0 $X86_TARGET|$STM32_TARGET|$ESP32C3_TARGET|$PARTICLE_ARGON_TARGET|$QEMU_CORTEX_A53|$QEMU_XTENSA_TARGET|$QEMU_RISCV64_TARGET|$QEMU_RISCV32_TARGET|$QEMU_ARC_TARGET"
```

Comment:
Not sure why the esp32 target is removed, is it because that esp32c3 is more popular and no need to support esp32 again, or here esp32c3 target is also suitable for esp32 target?

Reply:
Maybe this is a typo, esp32 is xtensa based, esp32c3 is risc-v based, they are different.

Reply:
If so, had better not remove esp32, including changes in src/main.c, boards/esp32.conf and esp32_custom_linker.ld. We had better support both.

Reply:
@wenyongh @no1wudi there are _many_ different esp32 variants and just `esp32` is not a valid target in Zephyr anymore, so build will fail. I would like to work on ensuring that all esp32 variants build correctly, but that will be a larger effort.

Reply:
@hasheddan got it, thanks, so we had better merge this PR first. Could you help enable the esp32 support after "Make Zephyr port work as a Zephyr module" as mentioned in #2782?

@no1wudi how do you think?

Reply:
> Could you help enable the esp32 support after "Make Zephyr port work as a Zephyr module" as mentioned in #2782?

@wenyongh absolutely!

Reply:
@wenyongh I'm fine with it

Reply:
OK, merged.